x-pack/plugins/security_solution/public/common/components/event_details/get_alert_summary_rows.tsx

```diff
@@ -17,88 +17,195 @@ import {
+      // resolve more fields based on the other event categories.
+      if (allEventCategories?.includes(EventCategory.FILE)) {
+        fields = fields.concat(getFieldsByCategory({ primaryEventCategory: EventCategory.FILE }));
+      }
 
-const memoryShellCodeAlertFields: EventSummaryField[] = [
-  ...defaultDisplayFields,
-  { id: 'rule.name', label: ALERTS_HEADERS_RULE_NAME },
-  {
-    id: 'Target.process.thread.Ext.start_address_details.memory_pe.imphash',
-    label: ALERTS_HEADERS_TARGET_IMPORT_HASH,
-  },
-];
+      if (allEventCategories?.includes(EventCategory.PROCESS)) {
+        fields = fields.concat(
+          getFieldsByCategory({ primaryEventCategory: EventCategory.PROCESS })
+        );
+      }
+      return fields;
+  }
+}
 
-const behaviorAlertFields: EventSummaryField[] = [
-  ...defaultDisplayFields,
-  { id: 'rule.description', label: ALERTS_HEADERS_RULE_DESCRIPTION },
-];
+/**
+ * Gets the fields to display based on the event's code.
+ * Contains some enhancements to resolve more fields based on the event's categories.
+ * @param eventCode The event's code
+ * @param eventCategories The events categories
+ * @returns A list of fields to include
+ */
+function getFieldsByEventCode(
+  eventCode: string | undefined,
+  eventCategories: EventCategories
+): EventSummaryField[] {
+  switch (eventCode) {
+    case EventCode.BEHAVIOR:
+      return [
+        { id: 'rule.description', label: ALERTS_HEADERS_RULE_DESCRIPTION },
+        // Resolve more fields based on the source event
+        ...getFieldsByCategory({ ...eventCategories, primaryEventCategory: undefined }),
+      ];
+    case EventCode.SHELLCODE_THREAD:
+      return [
+        { id: 'Target.process.executable' },
+        {
+          id: 'Target.process.thread.Ext.start_address_detaiuls.memory_pe.imphash',
+          label: ALERTS_HEADERS_TARGET_IMPORT_HASH,
+        },
+        {
+          id: 'Memory_protection.unique_key_v1',
+        },
+      ];
+    case EventCode.MEMORY_SIGNATURE:
+      return [{ id: 'process.Ext.memory_region.malware_signature.all_names' }];
+    case EventCode.RANSOMWARE:
+      // Resolve more fields based on the source event
+      return getFieldsByCategory({ ...eventCategories, primaryEventCategory: undefined });
+    default:
+      return [];
+  }
+}
 
-const memorySignatureAlertFields: EventSummaryField[] = [
-  ...defaultDisplayFields,
-  { id: 'rule.name', label: ALERTS_HEADERS_RULE_NAME },
-];
+/**
+ * Returns a list of fields based on the event's rule type
+ */
+function getFieldsByRuleType(ruleType?: string): EventSummaryField[] {
+  switch (ruleType) {
+    case 'threshold':
+      return [
+        { id: 'kibana.alert.threshold_result.count', label: ALERTS_HEADERS_THRESHOLD_COUNT },
+        { id: 'kibana.alert.threshold_result.terms', label: ALERTS_HEADERS_THRESHOLD_TERMS },
+        {
+          id: 'kibana.alert.threshold_result.cardinality',
```

Comment:
Also it would be good to use the constants for these `${ALERT_THRESHOLD_RESULT}.cardinality`, `${ALERT_THRESHOLD_RESULT}.terms`, etc...

Reply:
Ah cool, I was not aware of those -> https://github.com/elastic/kibana/pull/123239/commits/99d4d8d21ab26ea504156baa3ad8875b678b14af